### ISSUE TITLE
pydantic 2 version not always including charm source

### ISSUE THREAD

**Issue by lengau** [canonical/charmcraft#1812]
### Bug Description

the pydantic 2 release sometimes just includes metadata. Seems to happen for bases charms but not platforms ones

### To Reproduce

Run spread tests

### Environment

n/a

### charmcraft.yaml

```shell
n/a
```

### Relevant log output

```shell
2024-08-14 14:58:29 Error executing multipass:ubuntu-22.04-64:tests/spread/smoketests/basic:jammy_bases (multipass:ubuntu-22.04-64) : 
-----
+ cd charm
+ charmcraft pack --verbose
2024-08-14 14:57:54.271 Starting charmcraft, version 3.1.1.post65+g70078b0
2024-08-14 14:57:54.271 Logging execution to '/root/.local/state/charmcraft/log/charmcraft-20240814-145754.270839.log'
2024-08-14 14:57:54.271 Log verbosity level set to DEBUG
Starting charmcraft, version 3.1.1.post65+g70078b0
Logging execution to '/root/.local/state/charmcraft/log/charmcraft-20240814-145754.270839.log'
Launching managed ubuntu 22.04 instance...
Creating instance from base instance
Starting instance
Starting charmcraft, version 3.1.1.post65+g70078b0
Logging execution to '/tmp/charmcraft.log'
Starting charmcraft, version 3.1.1.post65+g70078b0
Logging execution to '/tmp/charmcraft.log'
Initialising lifecycle
Installing build-packages
Packing...
Packing charm jammy_ubuntu-22.04-amd64.charm
Packed jammy_ubuntu-22.04-amd64.charm
+ test -f jammy_ubuntu-22.04-amd64.charm
+ unzip -l jammy_ubuntu-22.04-amd64.charm
+ MATCH src/charm.py
grep error: pattern not found, got:
Archive:  jammy_ubuntu-22.04-amd64.charm
  Length      Date    Time    Name
---------  ---------- -----   ----
      271  2024-08-14 14:58   manifest.yaml
      156  2024-08-14 14:58   metadata.yaml
---------                     -------
      427                     2 files
-----
.
```

**Comment by lengau** [comment]
```
2024-08-14 15:37:07 Successful tasks: 40
2024-08-14 15:37:07 Aborted tasks: 1
2024-08-14 15:37:07 Failed tasks: 11
    - multipass:ubuntu-20.04-64:tests/spread/smoketests/basic:focal_bases
    - multipass:ubuntu-20.04-64:tests/spread/smoketests/basic:jammy_bases
    - multipass:ubuntu-20.04-64:tests/spread/smoketests/basic:noble_platforms
    - multipass:ubuntu-20.04-64:tests/spread/smoketests/metafiles-bad-yaml-unenforced
    - multipass:ubuntu-22.04-64:tests/spread/dependencies/requirements:basic
    - multipass:ubuntu-22.04-64:tests/spread/dependencies/requirements:hash
    - multipass:ubuntu-22.04-64:tests/spread/dependencies/requirements:hash_pydeps
    - multipass:ubuntu-22.04-64:tests/spread/dependencies/requirements:pydeps
    - multipass:ubuntu-22.04-64:tests/spread/smoketests/basic:focal_bases
    - multipass:ubuntu-22.04-64:tests/spread/smoketests/basic:jammy_bases
    - multipass:ubuntu-22.04-64:tests/spread/smoketests/metafiles-bad-yaml-unenforced
error: unsuccessful run
```

**Comment by lengau** [comment]
Fixed in #1815
